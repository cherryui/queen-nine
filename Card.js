import React from 'react'
import { View, StyleSheet } from 'react-native'
import { Avatar } from 'react-native-elements'
import Icons from './icons/Icons.js'

export default class Card extends React.Component {
  render () {
    // get icon based on props
    var icon = null
    switch (this.props.card) {
      case ('right'):
        icon = Icons.right
        break
      case ('left'):
        icon = Icons.left
        break
      case ('ace'):
        icon = Icons.ace
        break
      case ('king'):
        icon = Icons.king
        break
      case ('queen'):
        icon = Icons.queen
        break
      case ('ten'):
        icon = Icons.ten
        break
      case ('nine'):
        icon = Icons.nine
        break
    }

    // check if card has already been chosen -- if so, display check
    if (this.props.selected) {
      icon = Icons.success
    }
    return (
      <Avatar
        onPress={() => this.props.handleCardClick(this.props.card, this.props.selected)}
        containerStyle={styles.cardButton}
        source={icon}
        large
        rounded
      />
    )
  }
}

const styles = StyleSheet.create({
  cardButton: {
    marginRight: 10,
    marginLeft: 10
  }
})