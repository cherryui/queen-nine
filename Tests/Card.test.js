import 'react-native'
import React from 'react'
import Card from '../Components/Card'

import renderer from 'react-test-renderer'

test('renders correctly', () => {
  const component = renderer.create(
    <Card card='ace'/>
  ).toJSON()
  expect(component).toMatchSnapshot()
})
