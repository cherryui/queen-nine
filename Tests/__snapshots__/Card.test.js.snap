// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "transparent",
      "height": 75,
      "marginLeft": 10,
      "marginRight": 10,
      "opacity": 1,
      "paddingBottom": 10,
      "paddingRight": 10,
      "paddingTop": 10,
      "width": 75,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "rgba(0,0,0,0.2)",
          "bottom": 0,
          "flex": 1,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "borderRadius": 37.5,
        },
        undefined,
      ]
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "height": 75,
            "width": 75,
          },
          Object {
            "borderRadius": 37.5,
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;
